conjure-java-annotations/src/main/java/com/palantir/conjure/java/annotations/JerseyClient.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.annotations;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Indicator annotation for marking that a JAX-RS interface is used to construct a Jersey client.
+ * See also {@link JerseyService}.
+ */
+@Retention(RetentionPolicy.SOURCE)
+@Target(ElementType.TYPE)
+public @interface JerseyClient {}
```

Comment:
This is currently very specific to jersey. We might want to make this framework agnostic and name it something like `ClientDefinition` and `ServerDefinition`. But not sure if we have further use-cases outside the jersey migration yet.

Reply:
Maybe `JaxRsClient`? Jersey is a jax-rs implementation, and we don't use any of its client functionality. Feign is our jax-rs client-side implementation.

We could call the other annotation `JaxRsServer` for a clear client/server distinction.

Thoughts on using `RUNTIME` retention, that way our client factories can be updated to throw if an interface is annotated incorrectly?